src/sql/workbench/services/notebook/browser/interface.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as azdata from 'azdata';
+import { URI } from 'vs/base/common/uri';
+import { Event } from 'vs/base/common/event';
+import { IContentManager } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { IStandardKernelWithProvider } from 'sql/workbench/services/notebook/browser/models/notebookUtils';
+
+export interface INotebookInput {
+	defaultKernel: azdata.nb.IKernelSpec,
+	connectionProfile: azdata.IConnectionProfile,
+	isDirty(): boolean;
+	setDirty(boolean);
+	readonly notebookUri: URI;
+	updateModel(): void;
+	readonly editorOpenedTimestamp: number;
+	readonly layoutChanged: Event<void>;
+	readonly contentManager: IContentManager;
+	readonly standardKernels: IStandardKernelWithProvider[];
+}
+
+export function isINotebookInput(value: any): value is INotebookInput {
```

Comment:
I'm a bit confused by the return type here. Is this the same as having a return type of boolean?

Reply:
It's a special thing called a [type guard predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)

Basically it lets the compiler know whether something is the specified type (in this case INotebookInput) or not for purposes of verifying proper typing support. This is needed when the object isn't an actual class instance since interfaces/raw objects don't work with instanceof.